src/wifipaf/tests/TestWiFiPAFLayer.cpp

```diff
@@ -114,25 +114,31 @@ TEST_F(TestWiFiPAFLayer, CheckWiFiPAFTransportCapabilitiesResponseMessage)
 
 TEST_F(TestWiFiPAFLayer, CheckPafSession)
 {
-    const NodeId nodeId          = 0x1;
-    const uint16_t discriminator = 0xF00;
-    AddPafSession(nodeId, discriminator);
-
-    auto pPafSessionInfo_nodeid = GetPAFInfo(nodeId);
-    EXPECT_EQ(pPafSessionInfo_nodeid->nodeId, nodeId);
-    EXPECT_EQ(pPafSessionInfo_nodeid->discriminator, discriminator);
-
-    auto pPafSessionInfo_disc = GetPAFInfo(discriminator);
-    EXPECT_EQ(pPafSessionInfo_disc->nodeId, nodeId);
-    EXPECT_EQ(pPafSessionInfo_disc->discriminator, discriminator);
-
-    uint32_t pafid = 0x2;
-    AddPafSession(pafid);
-    auto pPafSessionInfo_Remote = GetPAFInfo(pafid);
-    EXPECT_EQ(pPafSessionInfo_Remote->id, pafid);
-    RmPafSession(pafid);
-    auto pPafSessionInfo_Remote_1 = GetPAFInfo(pafid);
-    EXPECT_EQ(pPafSessionInfo_Remote_1, nullptr);
+    // Add a session by giving node_id, discriminator
+    WiFiPAF::WiFiPAFSession sessionInfo = { .nodeId = 0x1, .discriminator = 0xF00 };
+    EXPECT_EQ(AddPafSession(PafInfoAccess::kAccNodeInfo, sessionInfo), CHIP_NO_ERROR);
+
+    // Get the session info by giving node_id
+    sessionInfo.nodeId          = 0x1;
+    auto pPafSessionInfo_nodeid = GetPAFInfo(PafInfoAccess::kAccNodeId, sessionInfo);
+    EXPECT_EQ(pPafSessionInfo_nodeid->nodeId, sessionInfo.nodeId);
+    EXPECT_EQ(pPafSessionInfo_nodeid->discriminator, sessionInfo.discriminator);
+
+    // Get the session info by giving the discriminator
+    sessionInfo.discriminator = 0xF00;
```

Comment:
- Please test for search of a session that does not exist
- Please test for the situation where there are > 1 PAF session and you look for a given discriminator (e.g. both first and second), and make sure you don't find the other one.

Reply:
Yes, I will add this and more cases to improve the coverage.